src/main/java/org/openstreetmap/atlas/checks/validation/intersections/EdgeCrossingEdgeCheck.java

```diff
@@ -105,22 +109,44 @@ public boolean validCheckForObject(final AtlasObject object)
         // edges from being flagged more than once.
         final Set<Edge> collectedEdges = new EdgeCrossingEdgeWalker((Edge) object,
```

Comment:
@Bentleysb Please take a look at this flag method and kindly let me know if these code changes are satisfying the functionality which we discussed.

Reply:
This logic looks like it should deal with most of the determinism issues. For extremely complex networks of crossing edges there will still be some nondeterminism and duplication in the flags, but I think that should be within reasonable levels. It would be good to have some numbers on how much duplication this would have.

Reply:
I updated the description on number of flags generated before and after. We can not completely eradicate duplicates for complex networks. But, this code definitely works in for taking care of duplicates by grouping the edges in one flag based on different conditions mentioned in the code.